### PR TITLE
Feature: cofl ah search item

### PR DESCRIPTION
**Pull request by hannibal002**
## What
Added an Item in AH search browser that allows searching for this item on https://sky.coflnet.com/

<details>
<summary>Images</summary>

image
opens https://sky.coflnet.com/item/HYPERION?range=day

</details>

## Changelog New Features
+ Added an option in the Auction House search browser to search for the item on coflnet.com. - hannibal2

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/inventory/auctionhouse/AuctionHouseOpenPriceWebsite.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        "Auctions: \"(?<searchTerm>.*)\""
+    )
+
+    private var searchTerm = ""
+    private var displayItem: ItemStack? = null
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!isEnabled()) return
+        cookieDurationPattern.matchMatcher(event.inventoryName) {
```
Comment (Ekwav) [inline src/main/java/at/hannibal2/skyhanni/features/inventory/auctionhouse/AuctionHouseOpenPriceWebsite.kt]: Looks like a copy paste error
```suggestion
    private val ahSearchPattern by patternGroup.pattern(
        "title.search",
        "Auctions: \"(?<searchTerm>.*)\""
    )

    private var searchTerm = ""
    private var displayItem: ItemStack? = null

    @SubscribeEvent
    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
        if (!isEnabled()) return
        ahSearchPattern.matchMatcher(event.inventoryName) {
```

File: src/main/java/at/hannibal2/skyhanni/config/features/inventory/AuctionHouseConfig.java [hunk 2]
```diff
@@ -49,4 +49,10 @@ public class AuctionHouseConfig {
     )
     @ConfigEditorKeybind(defaultKey = Keyboard.KEY_NONE)
     public int copyUnderbidKeybind = Keyboard.KEY_NONE;
+
+    @Expose
+    @ConfigOption(name = "Price Website", desc = "Adds a button to the Auction House that will open the item page in §ccoflnet.com§7.")
```
Comment (Ekwav) [inline src/main/java/at/hannibal2/skyhanni/config/features/inventory/AuctionHouseConfig.java]: To be consistent between both descriptions
```suggestion
    @ConfigOption(name = "Price Website", desc = "Adds a button to the Auction House that will open the item page in §csky.coflnet.com§7.")
```

**Review by Ekwav — APPROVED** [review]

File: src/main/java/at/hannibal2/skyhanni/config/features/inventory/AuctionHouseConfig.java (hunk 2 above)
Comment (Obsidianninja11) [inline src/main/java/at/hannibal2/skyhanni/config/features/inventory/AuctionHouseConfig.java]: Why is coflnet red?
Shouldn't it be yellow?
also what Ekwav said. sky.coflnet.com is the website

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/config/features/inventory/AuctionHouseConfig.java]: Yellow are only commands, red is important stuff. e.g. the name of an external service.

image

File: src/main/java/at/hannibal2/skyhanni/features/inventory/auctionhouse/AuctionHouseOpenPriceWebsite.kt [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    private var lastClick = SimpleTimeMark.farPast()
+
+    private val patternGroup = RepoPattern.group("inventory.auctionhouse")
+
+    /**
+     * REGEX-TEST: Auctions: "hyperion"
+     */
+    private val ahSearchPattern by patternGroup.pattern(
+        "title.search",
+        "Auctions: \"(?<searchTerm>.*)\""
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/inventory/auctionhouse/AuctionHouseOpenPriceWebsite.kt]: Does not match long patterns

image

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
just this

**Comment by hannibal002** [comment]
suddently this error happens
<details>
<summary>Stack trace</summary>

```
SkyHanni 0.25.Beta.26: Caught an URISyntaxException in AuctionHouseOpenPriceWebsite at SlotClickEvent: Illegal character in path at index 43: https://sky.coflnet.com/api/mod/open/Golden Dragon
 
Caused by java.net.URISyntaxException: Illegal character in path at index 43: https://sky.coflnet.com/api/mod/open/Golden Dragon
at java.net.URI$Parser.fail(URI.java:2848)
at java.net.URI$Parser.checkChars(URI.java:3021)
at java.net.URI$Parser.parseHierarchical(URI.java:3105)
at java.net.URI$Parser.parse(URI.java:3053)
at java.net.URI.<init>(URI.java:588)
at SH.utils.OSUtils.openBrowser(OSUtils.kt:16)
at SH.features.inventory.auctionhouse.AuctionHouseOpenPriceWebsite.onSlotClick(AuctionHouseOpenPriceWebsite.kt:83)
at SH.mixins.hooks.GuiContainerHook.onMouseClick(GuiContainerHook.kt:62)
at MC.client.gui.inventory.GuiContainer.handler$onMouseClick$zza000(GuiContainer.java:2307)
at MC.client.gui.inventory.GuiContainer.handleMouseClick(GuiContainer.java:629)
at MC.client.gui.inventory.GuiContainer.mouseClicked(GuiContainer.java:388)
at MC.client.gui.GuiScreen.handleMouseInput(GuiScreen.java:555)
at MC.client.gui.GuiScreen.handleInput(GuiScreen.java:524)
at MC.client.Minecraft.runTick(Minecraft.java:1674)
at MC.client.Minecraft.runGameLoop(Minecraft.java:1024)
at MC.client.Minecraft.run(Minecraft.java:349)
at MC.client.main.Main.main(Main.java:124)
```

</details>

when i try to fix it with this

<details>
<summary>Patch</summary>

```
Index: src/main/java/at/hannibal2/skyhanni/features/inventory/auctionhouse/AuctionHouseOpenPriceWebsite.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/at/hannibal2/skyhanni/features/inventory/auctionhouse/AuctionHouseOpenPriceWebsite.kt b/src/main/java/at/hannibal2/skyhanni/features/inventory/auctionhouse/AuctionHouseOpenPriceWebsite.kt
--- a/src/main/java/at/hannibal2/skyhanni/features/inventory/auctionhouse/AuctionHouseOpenPriceWebsite.kt(revision 0a5ccaa3d6be5113c8f4d6def9d966a1df720712)
+++ b/src/main/java/at/hannibal2/skyhanni/features/inventory/auctionhouse/AuctionHouseOpenPriceWebsite.kt(date 1715673321561)
@@ -17,6 +17,7 @@
 import net.minecraft.item.ItemStack
 import net.minecraftforge.fml.common.eventhandler.EventPriority
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.net.URLEncoder
 import kotlin.time.Duration.Companion.seconds
 
 class AuctionHouseOpenPriceWebsite {
@@ -79,7 +80,8 @@
         if (event.slotId != 8) return
         event.isCanceled = true
         if (lastClick.passedSince() > 0.3.seconds) {
-            val url = "https://sky.coflnet.com/api/mod/open/$searchTerm"
+            val search = URLEncoder.encode(searchTerm, "UTF-8")
+            val url = "https://sky.coflnet.com/api/mod/open/$search"
             OSUtils.openBrowser(url)
             lastClick = SimpleTimeMark.now()
         }
```


</details>
then the website links me to start page instead to the item page

**Review by Ekwav — APPROVED** [review]

**Review by CalMWolfs — APPROVED** [review]
code lgtm

**Pull request merged**